### PR TITLE
[t175] evaluate generated expressions on update

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java [hunk 1]
```diff
@@ -357,6 +378,69 @@ private IndexRequest prepareInsert(ShardUpsertRequest request, ShardUpsertReques
         return indexRequest;
     }
 
+    @VisibleForTesting
+    void processGeneratedColumns(final DocTableInfo tableInfo, Map<String, Object> updatedColumns) {
+        List<String> updatedColumnNames = Lists.newArrayList(updatedColumns.keySet());
+        List<ReferenceInfo> updatedReferenceInfos = Lists.transform(updatedColumnNames, new Function<String, ReferenceInfo>() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java]: the Function classes could probably be constants?

File: sql/src/main/java/io/crate/metadata/doc/DocTableInfo.java [hunk 2]
```diff
@@ -133,6 +136,12 @@ public DocTableInfo(TableIdent ident,
         // scale the fetchrouting timeout by n# of partitions
         this.routingFetchTimeout = new TimeValue(5 * Math.max(1, this.partitions.size()), TimeUnit.SECONDS);
         this.docColumn = new TableColumn(DocSysColumns.DOC, references);
+
+        for (ReferenceInfo referenceInfo : columns) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/metadata/doc/DocTableInfo.java]: unused ?

**Comment by mfussenegger** [comment]
otherwise ack

File: sql/src/main/java/io/crate/metadata/doc/DocTableInfo.java (hunk 2 above)
Comment (seut) [inline sql/src/main/java/io/crate/metadata/doc/DocTableInfo.java]: ah yes thx ;)

**Pull request merged**